apps/wfo-ui-surf/components/WfoServiceTicketDetailPage/utils.ts

```diff
@@ -1,13 +1,127 @@
-import { ServiceTicketProcessState } from '../../types';
-
-export enum ServiceTicketTabIds {
-    GENERAL_TAB = 'general-id',
-    NOTIFICATION_LOG = 'notification-log-id',
-    SENT_EMAILS = 'sent-emails-id',
-}
+import {
+    ImpactLevel,
+    ImpactLevelFromApi,
+    ImpactedCustomerRelation,
+    ImpactedCustomersTableColumns,
+    ImpactedObject,
+    MinlObjectFromApi,
+    ServiceTicketProcessState,
+} from '../../types';
 
 export const abortEnabledValues: ServiceTicketProcessState[] = [
     ServiceTicketProcessState.OPEN_RELATED,
     ServiceTicketProcessState.OPEN_ACCEPTED,
     ServiceTicketProcessState.OPEN,
 ];
+
+const impactLevelPriority = [
+    ImpactLevel.NO_IMPACT,
+    ImpactLevel.REDUCED_REDUNDANCY,
+    ImpactLevel.RESILIENCE_LOSS,
+    ImpactLevel.DOWN,
+];
+
+export const getImpactedObjectPriority = (impact: ImpactLevel) => {
+    return impactLevelPriority.indexOf(impact);
+};
+
+export const convertApiToUiImpactLevel = (
+    apiImpactLevel?: ImpactLevelFromApi,
+) => {
+    switch (apiImpactLevel) {
+        case ImpactLevelFromApi.NO_IMPACT:
+            return ImpactLevel.NO_IMPACT;
+        case ImpactLevelFromApi.REDUCED_REDUNDANCY:
+            return ImpactLevel.REDUCED_REDUNDANCY;
+        case ImpactLevelFromApi.RESILIENCE_LOSS:
+            return ImpactLevel.RESILIENCE_LOSS;
+        case ImpactLevelFromApi.DOWN:
+            return ImpactLevel.DOWN;
+        default:
+            return null;
+    }
+};
+
+export const getMinlForCustomer = (
+    customerId: string,
+    apiMinl: MinlObjectFromApi[],
+) => {
+    const impact = apiMinl.find((m) => m.customer_id === customerId)?.impact;
+    return convertApiToUiImpactLevel(impact);
+};
+
+export const calculateInformCustomer = (
+    sendingLevel: ImpactLevel,
+    acceptedImpact: ImpactLevel,
+) => {
+    return (
+        getImpactedObjectPriority(acceptedImpact) >=
+        getImpactedObjectPriority(sendingLevel)
+    );
+};
+
+export const calculateSendingLevel = (
+    minimalImpactLevel: ImpactLevel | null,
+    defaultSendingLevel: ImpactLevel,
+) => {
+    return minimalImpactLevel ? minimalImpactLevel : defaultSendingLevel;
```

Comment:
this could be `minimalImpactLevel ?? defaultSendingLevel` and the function without braces, almost a one-liner